develop/parachain-devs/runtime-development/FRAME/frame-overview.md

```diff
@@ -0,0 +1,110 @@
+
+#### Pallet Structure
+
+Polkadot SDK heavily utilizes Rust macros, allowing developers to focus on specific functional requirements when writing pallets instead of dealing with technicalities and scaffolding code.
+
+A typical pallet skeleton looks like this:
+
+```rust
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+  use frame_support::pallet_prelude::*;
+  use frame_system::pallet_prelude::*;
+
+  #[pallet::pallet]
+  #[pallet::generate_store(pub(super) trait Store)]
+  pub struct Pallet<T>(_);
+
+  #[pallet::config]  // snip
+  #[pallet::event]   // snip
+  #[pallet::error]   // snip
+  #[pallet::storage] // snip
+  #[pallet::call]    // snip
+}
+```
+
+All pallets, including custom ones, implement these attribute macros:
+
+- `#[frame_support::pallet]` - marks the module as usable in the runtime
+- `#[pallet::pallet]` - applied to a structure used to retrieve module information easily
+- `#[pallet::config]` - defines the configuration for the pallets's data types
+- `#[pallet::event]` - defines events to provide additional information to users
+- `#[pallet::error]` - lists possible errors in an enum to be returned upon unsuccessful execution
+- `#[pallet::storage]` - defines elements to be persisted in storage
+- `#[pallet::call]` - defines functions exposed as transactions, allowing dispatch to the runtime
+
+These macros are applied as attributes to Rust modules, functions, structures, enums, and types. They enable the pallet to be built and added to the runtime, exposing the custom logic to the outer world.
+
+!!! note
+    The macros above are the core components of a pallet. For a comprehensive guide on these and additional macros, refer to the [pallet_macros](https://paritytech.github.io/polkadot-sdk/master/frame_support/pallet_macros/index.html){target=\_blank} section in the Polkadot SDK documentation.
+
+### Support Libraries
+
+In addition to pallets, FRAME provides services to interact with the runtime through the following libraries and modules:
+
+- [`frame_system` crate]() - provides low-level types, storage, and functions for the runtime
+- [frame_support crate]() - is a collection of Rust macros, types, traits, and modules that simplify the development of Substrate pallets
+- [`frame_executive` pallet]() - orchestrates the execution of incoming function calls to the respective pallets in the runtime
+
+## Compose a Runtime with Pallets
+
+The Polkadot SDK allows developers to construct a runtime by combining various pallets, both built-in and custom-made. This modular approach enables the creation of unique blockchain behaviors tailored to specific requirements.
+
+The following diagram illustrates the process of selecting and combining FRAME pallets to compose a runtime:
+
+![](/images/develop/parachain-devs/runtime-development/FRAME/frame-overview/frame-overview-2.webp)
```

Comment:
Even though we will be launching only in light theme, I'd add to the post-migration tracking sheet that this image does not look very nice on dark backgrounds

Reply:
I added the dark image to the src files, but not sure how to switch between them when the them is changed. Probably @eshaben can help with that

Reply:
I wouldn't worry about adding in that logic in this PR, because we'll be switch to the light-only theme sooner than later.

But just so you know, this is how it would be done like this:

```
![](/images/develop/parachain-devs/runtime-development/FRAME/frame-overview/frame-overview-2.webp#only-light)
![](/images/develop/parachain-devs/runtime-development/FRAME/frame-overview/frame-overview-2.webp#only-dark)
```